src/app/shared/services/user-rate.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@angular/core';
+import {
+  Observable,
+  switchMap,
+  take,
+  tap
+} from 'rxjs';
+import { ApiRequestType } from '../enums/api-request';
+import { LocalStorageKeys } from '../enums/local-storage';
+import { MediaType } from '../enums/media-types';
+import {
+  GuestSession,
+  RatedCard
+} from '../interfaces/general';
+import { MovieShortCard } from '../interfaces/movies';
+import { ApiService } from './api.service';
+import { LocalStorageService } from './local-storage.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserRateService {
+  private _guestSession = '';
```

Comment:
In my opinion, `guestSession` is not related to UserRate functional. Maybe we could resurrect guest session service?